File: backend/app/schemas/stats.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class StatusCount(BaseModel):
    status: str
    count: int


class PriorityCount(BaseModel):
    priority: str
    count: int


class CategoryCount(BaseModel):
    category_id: uuid.UUID | None
    category_name: str
    count: int


class StatsOverviewResponse(BaseModel):
    by_status: list[StatusCount]
    by_priority: list[PriorityCount]
    by_category: list[CategoryCount]


class WorkloadItem(BaseModel):
    user_id: uuid.UUID
    first_name: str
    last_name: str
    active_ticket_count: int


class StatsWorkloadResponse(BaseModel):
    items: list[WorkloadItem]


class StatsOverviewFilterParams(BaseModel):
    department_ids: str | None = None
    date_from: datetime | None = None
    date_to: datetime | None = None


class StatsWorkloadFilterParams(BaseModel):
    date_from: datetime | None = None
    date_to: datetime | None = None
    workload_statuses: str | None = None

File: frontend/src/types/stats.ts
export interface StatusCount {
  status: string;
  count: number;
}

export interface PriorityCount {
  priority: string;
  count: number;
}

export interface CategoryCount {
  category_id: string | null;
  category_name: string;
  count: number;
}

export interface StatsOverviewResponse {
  by_status: StatusCount[];
  by_priority: PriorityCount[];
  by_category: CategoryCount[];
}

export interface WorkloadItem {
  user_id: string;
  first_name: string;
  last_name: string;
  active_ticket_count: number;
}

export interface StatsWorkloadResponse {
  items: WorkloadItem[];
}

export interface StatsOverviewFilterParams {
  department_ids?: string;
  date_from?: string;
  date_to?: string;
}

export interface StatsWorkloadFilterParams {
  date_from?: string;
  date_to?: string;
  workload_statuses?: string;
}
